src/ViewTopics.jsx

```diff
@@ -1,7 +1,45 @@
-export default function ViewTopics() {
+import { useState, useEffect } from "react";
+import Expandable from "./Expandable";
+import { Link } from "react-router-dom";
+import { getTopics } from "./UTILS/utils";
+import { capitalise } from "./UTILS/capitalise";
+
+import './CSS/nav.css'
+
+export default function ViewTopics({ searchParams, setSearchParams }) {
+  const [viewTopicTitles, setViewTopicTitles] = useState([]);
+  const [buttonsVisible, setButtonsVisible] = useState(false);
+
+  function setTopicQuery(topic) {
+    const newParams = new URLSearchParams(searchParams);
+    newParams.set("topic", topic);
+    setSearchParams(newParams);
+  }
+  useEffect(() => {
+    getTopics().then((response) => {
+      setViewTopicTitles(response.topics)
+      setButtonsVisible(true);
+    })
+    .catch((err) => {
+      console.log("error: ", err);
+    });
```

Comment:
Please add a loading pattern here as well, so it says loading while your topics are being loaded
https://notes.northcoders.com/courses/js-front-end/useEffect#loading-patterns